source/heroes/ControllableHero.hx

```diff
@@ -5,34 +5,85 @@ import flixel.util.FlxColor;
 import flixel.FlxG;
 
 class ControllableHero extends FlxSprite {
-    var DirectionX:Float;
-    var DirectionY:Float;
+    var HeroHeight:Int = 50;
+    var HeroWidth:Int = 25;
+    var HeroColor:FlxColor = 0xFF15A3BD;
+    var MaxSpeed:Int = 300;
+    var MinSpeed:Int = -300;
 
     public function new(X:Float=0, Y:Float=0)
     {
         super(X, Y);
-        makeGraphic(25, 50, 0xFF006118);
-
+        super.makeGraphic(HeroWidth, HeroHeight, HeroColor);
     }
     override public function update(elapsed:Float)
-    {
-        super.update(elapsed);
+    {        
+        if (this.velocity.y > 0 && this.y >= FlxG.height){
+            this.y = 0; 
+        }
+        else if (this.velocity.y < 0 && (this.y + this.height) <= 0){
+            this.y = FlxG.height;
+        }
         
+        if (this.velocity.x > 0 && this.x >= FlxG.width){
+            this.x = -(this.width);   
+        }
+        else if (this.velocity.x < 0 && (this.x + this.width) <= 0){
+            this.x = FlxG.width; 
+        }
+        
+        // Accelerate until max speed is reached
         if(FlxG.keys.pressed.UP)
         {
-            this.y -=5;
+            if(this.velocity.y >= MinSpeed)
+            {
+                this.velocity.y -= 10;
+            }
+            
         }
+
         if(FlxG.keys.pressed.DOWN)
         {
-            this.y +=5;
+            if(this.velocity.y <= MaxSpeed)
+            {
+                this.velocity.y += 10;
+            }
         }
+        trace(this.velocity.y);
+
         if(FlxG.keys.pressed.LEFT)
         {
-            this.x -=5;
+            if(this.velocity.x >= MinSpeed)
+            {
+                this.velocity.x -= 10;
+            }
         }
+       
         if(FlxG.keys.pressed.RIGHT)
         {
-            this.x +=5;
+            if(this.velocity.x <= MaxSpeed)
+            {
+                this.velocity.x += 10;
+            }
         }
+
+        // Decelerate if necessary
+        if(this.velocity.y > 0)
+        {
+            this.velocity.y -= 5;
+        }
+        if(this.velocity.y < 0)
+        {
+            this.velocity.y += 5;
+        }
+        if(this.velocity.x > 0)
+        {
+            this.velocity.x -= 5;
+        }
+        if(this.velocity.x < 0)
+        {
+            this.velocity.x += 5;
+        }
```

Comment:
It seems like the `drag` property might be an easier way to get what you're wanting here, it's supposed to constantly decelerate the velocity as long as no accelleration is actively being applied.

No change required, but worth looking at.